File: 02_Jay/26_sumdivisor.java
import java.util.ArrayList;

class Solution {
    public int solution(int n) {
        ArrayList<Integer> divisor = new ArrayList<Integer>();
        
        if (n==0){
            return 0;
        }
        for (int i=1;i<3001;i++){
            if (n%i == 0){
                divisor.add(n/i);
                divisor.add(i);
            }
        }
        int answer = divisor.stream().mapToInt(Integer::intValue).sum() / 2;
        return answer;
    }
}

File: 02_Jay/26_sumdivisor.py
def solution(n):
    divisor = []
    if n==0:
        return 0
        
    original = n
    for i in range(1,3001):
        
        if original%i == 0:
            divisor.append(original // i)
            divisor.append(i)
            
    return sum(divisor)//2